server/src/main/java/org/opensearch/indices/IndicesRequestCache.java

```diff
@@ -127,16 +133,48 @@ public final class IndicesRequestCache implements TieredCacheEventListener<Indic
             (k, v) -> k.ramBytesUsed() + v.ramBytesUsed()
         ).setMaximumWeight(sizeInBytes).setExpireAfterAccess(expire).build();
 
+        Function<BytesReference, QuerySearchResult> transformationFunction = (data) -> {
+            try {
+                return convertBytesReferenceToQSR(data);
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        };
+        // enabling this for testing purposes. Remove/tweak!!
+        long CACHE_SIZE_IN_BYTES = 1000000L;
+        String SETTING_PREFIX = "indices.request.cache";
+        String STORAGE_PATH = indicesService.getNodePaths()[0].indicesPath.toString() + "/request_cache";
+
+        EhCacheDiskCachingTier<Key, BytesReference> ehcacheDiskTier = new EhCacheDiskCachingTier.Builder<Key, BytesReference>()
+            .setKeyType(Key.class)
+            .setValueType(BytesReference.class)
+            .setExpireAfterAccess(TimeValue.MAX_VALUE)
+            .setSettings(settings)
+            .setThreadPoolAlias("ehcacheTest")
+            .setMaximumWeightInBytes(CACHE_SIZE_IN_BYTES)
+            .setStoragePath(STORAGE_PATH)
+            .setSettingPrefix(SETTING_PREFIX)
+            .setKeySerializer(new IRCKeyWriteableSerializer(this))
+            .setValueSerializer(new BytesReferenceSerializer())
+            .build();
+
         // Initialize tiered cache service. TODO: Enable Disk tier when tiered support is turned on.
-        tieredCacheService = new TieredCacheSpilloverStrategyService.Builder<Key, BytesReference>().setOnHeapCachingTier(
-            openSearchOnHeapCache
-        ).setTieredCacheEventListener(this).build();
+        tieredCacheService = new TieredCacheSpilloverStrategyService.Builder<Key, BytesReference, QuerySearchResult>()
+            .setOnHeapCachingTier(openSearchOnHeapCache)
+            .setOnDiskCachingTier(ehcacheDiskTier)
+            .setTieredCacheEventListener(this)
+            .withPreDiskCachingPolicyFunction(transformationFunction)
```

Comment:
Instead of passing this to TieredCacheSpilloverStrategyService, why not directly pass this inside DiskTierTookTimePolicy?
With below suggested change, this might be possible.